tests/integration_tests/Test/PM/Integration/Update.cls

```diff
@@ -0,0 +1,263 @@
+
+  do ..ModuleNormalUpdateNoMirrorHelper(..#ModuleNoDependencies, updateStepsToSeed, updateStepsToRun, ..#ModuleOneVersion, ..#ModuleTwoVersion, 0)
+  do ..ModuleNormalUpdateNoMirrorHelper(..#ModuleNoDependencies, updateStepsToSeed, updateStepsToRun, ..#ModuleOneVersion, ..#ModuleTwoVersion, 1)
+}
+
+Method TestUpdateToOlderVersion()
+{
+  // Load module
+  do ..AssertModuleInstalledCorrectly(..#ModuleNoDependencies, ..#ModuleTwoVersion)
+
+  // Update module
+  set tSC = ##class(%IPM.Main).Shell("update -v "_..#ModuleNoDependencies_" "_..#ModuleOneVersion)
+  do $$$AssertStatusNotOK(tSC, "Attempt at updating module to older version failed.")
+}
+
+Method TestUpdateToInvalidVersion()
+{
+  // Load module
+  do ..AssertModuleInstalledCorrectly(..#ModuleNoDependencies, ..#ModuleOneVersion)
+
+  // Update module
+  set tSC = ##class(%IPM.Main).Shell("update -v "_..#ModuleNoDependencies_" 1.0.0")
+  do $$$AssertStatusNotOK(tSC, "Attempt at updating module to non-existent version failed.")
+}
+
+Method TestUpdateToNewVersionTwice()
+{
+  // Load module
+  do ..AssertModuleInstalledCorrectly(..#ModuleNoDependencies, ..#ModuleOneVersion)
+
+  // Update module to a 2.x version
+  set tSC = ##class(%IPM.Main).Shell("update -v "_..#ModuleNoDependencies_" "_..#ModuleTwoVersion)
+  do $$$AssertStatusOK(tSC, "Updated module to "_..#ModuleTwoVersion_" without error")
+
+  // Update module to a 3.x version
+  set tSC = ##class(%IPM.Main).Shell("update -v "_..#ModuleNoDependencies_" "_..#ModuleThreeVersion)
+  do $$$AssertStatusOK(tSC, "Updated module to "_..#ModuleThreeVersion_" without error")
+}
+
+Method TestUpdateToSameVersionTwiceNoMirror()
+{
+  // Load module
+  do ..AssertModuleInstalledCorrectly(..#ModuleNoDependencies, ..#ModuleOneVersion)
+
+  // Update module to a 2.x version
+  set tSC = ##class(%IPM.Main).Shell("update -v "_..#ModuleNoDependencies_" "_..#ModuleTwoVersion)
+  do $$$AssertStatusOK(tSC, "Updated module to "_..#ModuleTwoVersion_"without error")
+
+  // Get timestamps of the V2 methods
+  set stepTwoA = ..GetUpdateStep("Update.Update.V2", "MethodA", 1)
+  set stepTwoB = ..GetUpdateStep("Update.Update.V2", "MethodB", 0)
+  set stepTwoC = ..GetUpdateStep("Update.Update.V2", "MethodC", 1)
+
+  set twoAStart = stepTwoA.TimeStampStart
+  set twoAEnd = stepTwoA.TimeStampEnd
+
+  set twoBStart = stepTwoB.TimeStampStart
+  set twoBEnd = stepTwoB.TimeStampEnd
+
+  set twoCStart = stepTwoC.TimeStampStart
+  set twoCEnd = stepTwoC.TimeStampEnd
+
+  // Update module to SAME version
+  set tSC = ##class(%IPM.Main).Shell("update -v "_..#ModuleNoDependencies_" "_..#ModuleTwoVersion)
+  do $$$AssertStatusOK(tSC, "Updated module to "_..#ModuleTwoVersion_"without error")
+
+  // Check that second update does not rerun update steps
+  // (timestamps on each update step should not have changed)
+  do $$$AssertEquals(twoAStart, stepTwoA.TimeStampStart)
```

Comment:
After the update, you will want to call GetUpdateStep again to force reload of the persistence or else it will use the one from the first call (can do so using %Reload() as well instead as well instead of calling GetUpdateStep() again)